osu.Game/Screens/Select/BeatmapCarousel.cs

```diff
@@ -281,32 +293,26 @@ private void beatmapsChanged(IRealmCollection<BeatmapInfo> sender, ChangeSet cha
         {
             Guid? previouslySelectedID = null;
 
-            CarouselBeatmapSet existingSet = beatmapSets.FirstOrDefault(b => b.BeatmapSet.Equals(beatmapSet));
-
             // If the selected beatmap is about to be removed, store its ID so it can be re-selected if required
-            if (existingSet?.State?.Value == CarouselItemState.Selected)
+            if (selectedBeatmapSet?.BeatmapSet.ID == beatmapSet.ID)
                 previouslySelectedID = selectedBeatmap?.BeatmapInfo.ID;
 
             var newSet = createCarouselSet(beatmapSet);
 
-            if (existingSet != null)
-                root.RemoveChild(existingSet);
+            root.RemoveChild(beatmapSet.ID);
 
-            if (newSet == null)
+            if (newSet != null)
```

Comment:
Bit of an interesting behavioural change here - the `newSet == null` case didn't use to trigger `BeatmapSetsChanged()` (by early-returning and skipping that event's invocation at the end of this method) - but I think this new version's actually *more* correct? Looks like that branch is specific to cases wherein all difficulties in a set are hidden, so probably not a huge deal either way as I'd like to think that should be rather rare.

Reply:
Yeah, I figured this was more correct.